gson/src/main/java/com/google/gson/stream/JsonReader.java

```diff
@@ -292,55 +293,94 @@ public JsonReader(Reader in) {
-   *   <li>Array elements separated by {@code ;} instead of {@code ,}.
-   *   <li>Unnecessary array separators. These are interpreted as if null
-   *       was the omitted value.
-   *   <li>Names and values separated by {@code =} or {@code =>} instead of
-   *       {@code :}.
-   *   <li>Name/value pairs separated by {@code ;} instead of {@code ,}.
-   * </ul>
-   *
-   * <p>Note: Even in strict mode there are slight derivations from the JSON
-   * specification:
-   * <ul>
-   *   <li>JsonReader allows the literals {@code true}, {@code false} and {@code null}
-   *       to have any capitalization, for example {@code fAlSe}
-   *   <li>JsonReader supports the escape sequence {@code \'}, representing a {@code '}
-   *   <li>JsonReader supports the escape sequence <code>\<i>LF</i></code> (with {@code LF}
-   *       being the Unicode character U+000A), resulting in a {@code LF} within the
-   *       read JSON string
-   *   <li>JsonReader allows unescaped control characters (U+0000 through U+001F)
-   * </ul>
+   * @param lenient provoding <code>true</code> will set the strictness of this reader to {@link Strictness#LENIENT} and
+   *                passing <code>false</code> will set the strictness of this reader to {@link Strictness#DEFAULT}.
+   * @see #setStrictness(Strictness)  
    */
   public final void setLenient(boolean lenient) {
-    this.lenient = lenient;
+    this.strictness = lenient ? Strictness.LENIENT : Strictness.DEFAULT;
   }
 
   /**
-   * Returns true if this parser is liberal in what it accepts.
+   * Returns true if the {@link Strictness} of this reader is equal to {@link Strictness#STRICT}.
+   *
+   * @see #setStrictness(Strictness)
    */
   public final boolean isLenient() {
-    return lenient;
+    return strictness == Strictness.LENIENT;
   }
 
+  /**
+   * Configure how liberal this parser is in what it accepts.
+   *
+   * <p>In {@linkplain Strictness#STRICT strict} mode, the
+   * parser only accepts JSON in accordance with <a href="http://www.ietf.org/rfc/rfc4627.txt">RFC 4627</a>.
+   * {@linkplain Strictness#LENIENT lenient} mode, all sort of non-spec compliant JSON is accepted (see below).
+   * In {@linkplain Strictness#DEFAULT default} mode, only JSON in accordance with <a
+   * href="http://www.ietf.org/rfc/rfc4627.txt">RFC 4627</a> is accepted, with a few exceptions denoted below
+   * for backwards compatibility reasons.</p>
+   *
+   * <ul>
+   *     <li>
+   *         {@link Strictness#LENIENT}: In lenient mode, the following syntax erros are accepted:
+   *         <ul>
+   *             <li>Streams that start with the <a href="#nonexecuteprefix">non-execute prefix</a>, <code>")]}'\n"
+   *                 </code>.
+   *             <li>Streams that include multiple top-level values. With default or strict parsing,
+   *                 each stream must contain exactly one top-level value.
+   *             <li>Numbers may be {@link Double#isNaN() NaNs} or {@link Double#isInfinite() infinities}.
```

Comment:
I think we should say how these are spelled in our pseudo-JSON.

Reply:
I added that they are represented as `NaN` and `(-)Infinity` respectively